src/main/java/school/hei/haapi/service/FeeService.java

```diff
@@ -1,135 +1,86 @@
-        .filter(fee -> fee.getStatus().equals(status))
-        .sorted(Comparator.comparing(Fee::getDueDatetime).reversed())
-        .collect(Collectors.toUnmodifiableList());
-    if (firstIndex >= feesByStatus.size()) {
-      return List.of();
-    }
-    if (lastIndex > feesByStatus.size() - 1) {
-      lastIndex = feesByStatus.size();
-    }
-    return feesByStatus.subList(firstIndex, lastIndex);
-  }
-
-  private school.hei.haapi.endpoint.rest.model.Fee.StatusEnum getFeeStatus(Fee fee) {
-    if (computeRemainingAmount(fee) == 0) {
-      return school.hei.haapi.endpoint.rest.model.Fee.StatusEnum.PAID;
-    } else {
-      if (Instant.now().isAfter(fee.getDueDatetime())) {
-        return school.hei.haapi.endpoint.rest.model.Fee.StatusEnum.LATE;
-      }
-      return school.hei.haapi.endpoint.rest.model.Fee.StatusEnum.UNPAID;
+  public List<Fee> getFeesByStudentId(String studentId, PageFromOne page, BoundedPageSize pageSize,
+                                      school.hei.haapi.endpoint.rest.model.Fee.StatusEnum status) {
+    Pageable pageable =
+        PageRequest.of(page.getValue() - 1, pageSize.getValue(), Sort.by(DESC, "dueDatetime"));
+    if (status != null) {
+      return feeRepository.getFeesByStudentIdAndStatus(studentId, status, pageable);
     }
+    return feeRepository.getByStudentId(studentId, pageable);
   }
 
-  private int computeRemainingAmount(Fee fee) {
-    List<Payment> payments = fee.getPayments();
-    if (payments != null) {
-      int amount = payments
-          .stream()
-          .mapToInt(Payment::getAmount)
-          .sum();
-      return fee.getTotalAmount() - amount;
+  public Fee updateFeeStatus(Fee initialFee) {
+    if (initialFee.getRemainingAmount() == 0) {
+      initialFee.setStatus(PAID);
+    } else if (Instant.now().isAfter(initialFee.getDueDatetime())) {
+      initialFee.setStatus(LATE);
     }
-    return fee.getTotalAmount();
-  }
-
-  private Fee resetPaymentRelatedInfo(Fee initialFee) {
-    return Fee.builder()
-        .id(initialFee.getId())
-        .student(initialFee.getStudent())
-        .type(initialFee.getType())
-        .remainingAmount(computeRemainingAmount(initialFee))
-        .status(getFeeStatus(initialFee))
-        .comment(initialFee.getComment())
-        .creationDatetime(initialFee.getCreationDatetime())
-        .dueDatetime(initialFee.getDueDatetime())
-        .payments(initialFee.getPayments())
-        .totalAmount(initialFee.getTotalAmount())
-        .build();
+    return initialFee;
   }
 
-  private List<Fee> resetPaymentRelatedInfo(List<Fee> fees) {
-    return fees.stream()
-        .map(this::resetPaymentRelatedInfo)
-        .collect(toUnmodifiableList());
+  @Scheduled(cron = "0 0 * * * *")
+  public void updateFeesStatusToLate() {
+    List<Fee> unpaidFees = feeRepository.getUnpaidFees();
+    unpaidFees.forEach(fee -> {
+      updateFeeStatus(fee);
+      log.info("Fee with id." + fee.getId() + " was updated from UNPAID to LATE");
```

Comment:
Because this method is before the update, reformat the message as.

was udpated -> is going to be updated